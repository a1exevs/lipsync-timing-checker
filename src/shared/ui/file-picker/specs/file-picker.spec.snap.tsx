import { render } from '@testing-library/react';
import { FileAudio } from 'lucide-react';
import React from 'react';

import FilePicker from 'src/shared/ui/file-picker/file-picker';

describe('FilePicker snapshots', () => {
  it('renders default', () => {
    const { asFragment } = render(<FilePicker text="Upload" handleFileUpload={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with icon', () => {
    const { asFragment } = render(<FilePicker text="Upload" handleFileUpload={jest.fn()} icon={<FileAudio />} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with fileName', () => {
    const { asFragment } = render(<FilePicker text="Upload" handleFileUpload={jest.fn()} fileName="test.txt" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders disabled', () => {
    const { asFragment } = render(<FilePicker text="Upload" handleFileUpload={jest.fn()} disabled />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with accept prop', () => {
    const { asFragment } = render(
      <FilePicker text="Upload" handleFileUpload={jest.fn()} accept=".txt" fileInputTestId="file-input" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
