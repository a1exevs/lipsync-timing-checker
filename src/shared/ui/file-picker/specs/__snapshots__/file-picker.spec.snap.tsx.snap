// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FilePicker snapshots renders default 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row gap-2 items-center"
  >
    <button
      class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white w-1/2"
      type="button"
    >
      <span>
        Upload
      </span>
    </button>
    <input
      accept="*/*"
      style="display: none;"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`FilePicker snapshots renders disabled 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row gap-2 items-center"
  >
    <button
      class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white w-1/2"
      disabled=""
      type="button"
    >
      <span>
        Upload
      </span>
    </button>
    <input
      accept="*/*"
      style="display: none;"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`FilePicker snapshots renders with accept prop 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row gap-2 items-center"
  >
    <button
      class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white w-1/2"
      type="button"
    >
      <span>
        Upload
      </span>
    </button>
    <input
      accept=".txt"
      data-testid="file-input"
      style="display: none;"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`FilePicker snapshots renders with fileName 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row gap-2 items-center"
  >
    <button
      class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white w-1/2"
      type="button"
    >
      <span>
        Upload
      </span>
    </button>
    <span
      class="flex-1 block truncate text-sm text-gray-300"
      title="test.txt"
    >
      test.txt
    </span>
    <input
      accept="*/*"
      style="display: none;"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`FilePicker snapshots renders with icon 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row gap-2 items-center"
  >
    <button
      class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white w-1/2"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          class="lucide lucide-file-audio"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.5 22h.5a2 2 0 0 0 2-2V7l-5-5H6a2 2 0 0 0-2 2v3"
          />
          <path
            d="M14 2v4a2 2 0 0 0 2 2h4"
          />
          <path
            d="M2 19a2 2 0 1 1 4 0v1a2 2 0 1 1-4 0v-4a6 6 0 0 1 12 0v4a2 2 0 1 1-4 0v-1a2 2 0 1 1 4 0"
          />
        </svg>
      </span>
      <span>
        Upload
      </span>
    </button>
    <input
      accept="*/*"
      style="display: none;"
      type="file"
    />
  </div>
</DocumentFragment>
`;
